test/features/support/oauth_token.js

```diff
@@ -0,0 +1,234 @@
+const { stash, spec } = require("pactum");
+const chai = require("chai");
+const { like, regex } = require("pactum-matchers");
+const { Given, When, Then, Before, After } = require("@cucumber/cucumber");
+const { localhost } = require("./helpers/helpers");
+
+const jsonToBase64 = (jsonObj) => {
+  const jsonString = JSON.stringify(jsonObj);
+  return Buffer.from(jsonString).toString("base64");
+};
+
+const base64ToJson = (base64String) => {
+  const jsonString = Buffer.from(base64String, "base64").toString();
+  return JSON.parse(jsonString);
+};
+
+// client_assertion is a string in JWT format that has required claims: iss, sub, aud, exp, iat
+stash.addDataTemplate({
+  ValidRequestBody: {
+    grant_type: "authorization_code",
+    client_assertion_type:
+      "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+    client_assertion:
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9." +
+      jsonToBase64({
+        iss: "string",
+        sub: "string",
+        aud: "string",
+        exp: 2016240000,
+        iat: 1516239022,
+        name: "string",
+      }) +
+      ".C5YtmxEhZfc0ZXmBv9msTPDmKnnLK0d5esho3CB_we8",
+    client_id: "string",
+    code: "string",
+    redirect_uri: "http://example.com",
+  },
+});
+
+let specOAuthToken;
+
+const baseUrl = `${localhost}oauth/token`;
+
+let errorResultFunction = async (errorMessage) => {
+  specOAuthToken.expectStatus(400).expectJsonMatch({
+    error: errorMessage,
+  });
+
+  await specOAuthToken.toss();
+};
+
+Before(() => {
+  specOAuthToken = spec().expectResponseTime(15000);
+});
+
+// Background: The user succesfully recieves the ID and access token using the OAuth token endpoint
+Given(
+  "The user wants to recieve the ID and access token using the OAuth token endpoint",
+  () => {
+    return "The user wants to recieve the ID and access token using the OAuth token endpoint";
+  }
+);
+
+// Scenario: The user succesfully recieves the ID and access token using the OAuth token endpoint
+When(
+  "The user triggers an action to recieve the ID and access token with the valid payload",
+  () => {
+    // add .inspect() to see request/response body
+    specOAuthToken
+      .post(baseUrl)
+      .withJson({ "@DATA:TEMPLATE@": "ValidRequestBody" });
+  }
+);
+
+Then(
+  "The user succesfully recieves the ID and access token using the OAuth token endpoint",
+  async () => {
+    specOAuthToken
+      .expectStatus(200)
+      .expectJsonMatch({
+        id_token: regex(
+          "string.string.string",
+          /^([a-zA-Z0-9_=]+)\.([a-zA-Z0-9_=]+)\.([a-zA-Z0-9_\-\+\/=]*)/
```

Comment:
This will accept 'string.string.' without the third encoded part but in JWT it's never empty - it combines header, payload and a secret key encoded together.
Is there any reason why it allows zero characters there?

```suggestion
          /^([a-zA-Z0-9_=]+)\.([a-zA-Z0-9_=]+)\.([a-zA-Z0-9_\-\+\/=]+)/
```

Reply:
You're right, thanks.
No reasons there.